src/dex/balancer-v2/balancer-v2.ts

```diff
@@ -404,6 +497,20 @@ export class BalancerV2EventPool extends StatefulEventSubscriber<PoolStateMap> {
     let i = 0;
     const onChainStateMap = subgraphPoolBase.reduce(
       (acc: { [address: string]: PoolState }, pool) => {
+        if (['AaveLinear'].includes(pool.poolType)) {
+          // This will decode multicall data for all pools associated with linear pool
+          const [decoded, newIndex] = LinearPool.decodeOnChainCalls(
+            pool,
+            this.poolInterfaces,
+            this.vaultInterface,
+            data,
+            i,
+          );
+          i = newIndex;
+          acc = { ...acc, ...decoded };
+          return acc;
+        }
+
```

Comment:
May be it would be better to refactor the the getOnChainState logic for all types of pools to have `getOnChainCalls`, `decodeOnChainCalls` functions in the pool class similar to what you did for linear pools. Also the poolInterface can be simply be the member of the poolClass so the interface is nor required by the `BalancerV2EventPool`.

Reply:
I've refactored as you suggested and also added a parsePoolPairData helper for each pool type as I thought it helped keep things more organised.